Content.Client/GameObjects/Components/Power/SolarControlConsoleBoundUserInterface.cs

```diff
@@ -0,0 +1,233 @@
+                rows.AddChild(new Label {Text = "°"});
+
+                rows.AddChild(new Label {Text = "Panel Angular Velocity:"});
+                rows.AddChild(new Label {Text = ""});
+
+                rows.AddChild(PanelVelocity = new LineEdit());
+                rows.AddChild(new Label {Text = "°/min."});
+
+                rows.AddChild(new Label {Text = "Press Enter to confirm."});
+                rows.AddChild(new Label {Text = ""});
+
+                PanelRotation.SizeFlagsHorizontal = SizeFlags.FillExpand;
+                PanelVelocity.SizeFlagsHorizontal = SizeFlags.FillExpand;
+                rows.SizeFlagsHorizontal = SizeFlags.Fill;
+                rows.SizeFlagsVertical = SizeFlags.Fill;
+
+                NotARadar = new SolarControlNotARadar(igt);
+
+                var outerColumns = new HBoxContainer();
+                outerColumns.AddChild(rows);
+                outerColumns.AddChild(NotARadar);
+                outerColumns.SizeFlagsHorizontal = SizeFlags.Fill;
+                outerColumns.SizeFlagsVertical = SizeFlags.Fill;
+                Contents.AddChild(outerColumns);
+                Resizable = false;
+            }
+        }
+
+        private sealed class SolarControlNotARadar : Control
+        {
+            // IoC doesn't apply here, so it's propagated from the parent class.
+            // This is used for client-side prediction of the panel rotation.
+            // This makes the display feel a lot smoother.
+            private IGameTiming _gameTiming;
+
+            private SolarControlConsoleBoundInterfaceState _lastState = new SolarControlConsoleBoundInterfaceState(0, 0, 0, 0);
+
+            private TimeSpan _lastStateTime = TimeSpan.Zero;
+
+            public const int SizeFull = 290;
+            public const int RadiusCircle = 140;
+
+            public SolarControlNotARadar(IGameTiming igt)
+            {
+                _gameTiming = igt;
+            }
+
+            public void UpdateState(SolarControlConsoleBoundInterfaceState ls)
+            {
+                _lastState = ls;
+                _lastStateTime = _gameTiming.CurTime;
+            }
+
+            protected override Vector2 CalculateMinimumSize()
+            {
+                return (SizeFull, SizeFull);
+            }
+
+            protected override void Draw(DrawingHandleScreen handle)
+            {
+                int point = SizeFull / 2;
+                Color fakeAA = new Color(0.08f, 0.08f, 0.08f);
+                Color gridLines = new Color(0.08f, 0.08f, 0.08f);
+                int panelExtentCutback = 4;
+                int gridLinesRadial = 8;
+                //int gridLinesEquatorial = 8;
+
+                // Draw base
+                handle.DrawCircle((point, point), RadiusCircle + 1, fakeAA);
+                handle.DrawCircle((point, point), RadiusCircle, Color.Black);
+
+                // Draw grid lines
+                /* Requires RobustToolbox#1084
```

Comment:
Was merged.

Reply:
Handled (this is why the PR is failing CI, RobustToolbox submodule was not updated and docs say not to update it in the PR)